packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:html' as html;
+
+import 'package:google_sign_in_platform_interface/google_sign_in_platform_interface.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:js/js.dart';
+import 'package:meta/meta.dart';
+
+import 'src/generated/gapiauth2.dart' as auth2;
+// TODO: Remove once this lands https://github.com/dart-lang/language/issues/671
+import 'src/generated/gapiauth2.dart' show GoogleAuthExtensions;
+import 'src/load_gapi.dart' as gapi;
+import 'src/utils.dart' show gapiUserToPluginUserData;
+
+const String _kClientIdMetaSelector = 'meta[name=google-signin-client_id]';
+const String _kClientIdAttributeName = 'content';
+
+@visibleForTesting
+String gapiUrl = 'https://apis.google.com/js/platform.js';
+
+/// Implementation of the google_sign_in plugin for Web
+class GoogleSignInPlugin extends GoogleSignInPlatform {
+  GoogleSignInPlugin() {
+    _autoDetectedClientId = html
+        .querySelector(_kClientIdMetaSelector)
+        ?.getAttribute(_kClientIdAttributeName);
+
+    _isGapiInitialized = gapi.inject(gapiUrl).then((_) => gapi.init()).then((_) {
+      isInitialized = true;
+    });
+  }
+
+  Future<void> _isGapiInitialized;
+
+  @visibleForTesting
+  Future<void> get isInitializing => _isGapiInitialized;
+  @visibleForTesting
+  bool isInitialized;
+
+  String _autoDetectedClientId;
+  FutureOr<auth2.GoogleUser> _lastSeenUser;
+
+  static void registerWith(Registrar registrar) {
+    GoogleSignInPlatform.instance = GoogleSignInPlugin();
+  }
+
+  @override
+  Future<void> init(
+      {@required String hostedDomain,
+      List<String> scopes = const <String>[],
+      SignInOption signInOption = SignInOption.standard,
+      String clientId}) async {
+    final String appClientId = clientId ?? _autoDetectedClientId;
+    assert(
+        appClientId != null,
+        'ClientID not set. Either set it on a '
+        '<meta name="google-signin-client_id" content="CLIENT_ID" /> tag,'
+        ' or pass clientId when calling init()');
+
+    await _isGapiInitialized;
+
+    // This init returns an  user, so let's wait for its future
```

Comment:
I'm confused by this comment. I don't see any Future being awaited here

Reply:
Haha, I changed my mind on how to implement this, but never updated this comment. Thanks!